src/main/java/lotto/model/WinningLotto.java

```diff
@@ -0,0 +1,43 @@
+package lotto.model;
+
+import java.util.Set;
+
+public class WinningLotto {
+    private static final String BONUS_BALL_CONTAIN_LOTTO_NUMBER_MESSAGE = "로또넘버와 보너스볼은 중첩 될 수 없습니다.";
+
+    private final Lotto winningLotto;
+    private final LottoNumber bonusBall;
+
+    public WinningLotto(Set<LottoNumber> lottoNumbers, LottoNumber bonusBall) {
+        validateBonusBallNotContainLottoNumbers(lottoNumbers, bonusBall);
+        this.winningLotto = new Lotto(lottoNumbers);
+        this.bonusBall = bonusBall;
+    }
+
+    private void validateBonusBallNotContainLottoNumbers(Set<LottoNumber> lottoNumbers, LottoNumber bonusBall) {
+        if (lottoNumbers.contains(bonusBall)) {
+            throw new IllegalArgumentException(BONUS_BALL_CONTAIN_LOTTO_NUMBER_MESSAGE);
+        }
+    }
+
+    public WinningState makeWinningState(Lotto lotto) {
+        int winningCount = makeWinningCount(lotto.getLottoNumbers(), getLottoNumbers());
+        WinningState winningState = new WinningState(winningCount, checkBonusCount(lotto, this.bonusBall));
+
+        return winningState;
+    }
+
+    private int makeWinningCount(Set<LottoNumber> lottoNumbers, Set<LottoNumber> winningLottoNumbers) {
+        return (int) lottoNumbers.stream()
+                .filter((lottoNumber) -> winningLottoNumbers.contains(lottoNumber))
+                .count();
```

Comment:
```suggestion
    private int makeWinningCount(Set<LottoNumber> lottoNumbers) {
        return (int) lottoNumbers.stream()
                .filter((lottoNumber) -> this.winningLotto.contain(lottoNumber))
                .count();
```

이미 `WinningLotto`의 상태 값으로 `winningLotto`를 가지고 있기 때문에 외부에서 winningLottoNumbers를 받아올 필요 없을 것 같아요! 😃

마찬가지로 이 메서드를 호출하는 `makeWinningState` 에서도 `lotto`와 `winningLotto`끼리 비교하여 `같은 로또 숫자 개수가 몇 개인지를 파악하라!`와 같은 메시지를 보내도록 할 수 있을 것 같아요. (Lotto와 Lotto 타입의 비교가 되겠네요 🤔)